docs/cli/migration/list-nullability.md

```diff
@@ -0,0 +1,192 @@
+
+```
+type ListStringContainer @model {
+  stringList: [String!]!
+  stringNullableList: [String!]
+  nullableStringList: [String]!
+  nullableStringNullableList: [String]
+}
+```
+
+In this example, there are four fields with different combinations of optionality:
+- `stringList` - the list component and the list is required
+- `stringNullableList` - the list component is required, the list is optional
+- `nullableStringList` - the list component is optional, the list is required
+- `nullableStringNullableList` - the list component and list is optional
+
+> The list component in this example is a String type, however, this applies for other types as well such as Int, Bool, and embedded types that you define yourself.
+
+### **Why are we introducing this change?** 
+
+This is to align the optionality of the generated Swift models as closely as possible to the type defined in the schema.
+
+### **Who is impacted?** 
+
+<amplify-block-switcher>
+
+<amplify-block name="iOS">
+
+Developers building an iOS app with Amplify DataStore or Amplify API generates Swift models by running the command `amplify codegen models`. 
+
+ _Previous generated Swift code_
+
+```swift
+public struct ListStringContainer: Model {
+  public var stringList: [String]
+  public var stringNullableList: [String]
+  public var nullableStringList: [String]?
+  public var nullableStringNullableList: [String]?
+  ...
+}
+```
+
+_Current code generated Swift model_
+
+```swift
+public struct ListStringContainer: Model {
+  public var stringList: [String]
+  public var stringNullableList: [String]?
+  public var nullableStringList: [String?]
+  public var nullableStringNullableList: [String?]?
+  ...
+}
+```
+
+The difference between the current and previous code:
+
+- `stringList` - No changes
+- `stringNullableList` - the list was required and is now optional.
+- `nullableStringList` - the list component was required and is now optional. The list was optional and is now required
+- `nullableStringNullableList` - the list component was required and is now optional. 
+
+</amplify-block>
+
+</amplify-block-switcher>
+
+
+### **When do I have to upgrade?**
+
+This is behind a feature flag in Amplify CLI x.y.z and will be deprecated by [TODO: DATE]. Developers with existing apps should upgrade to the latest CLI, set the feature flag, and update their app code to account for the change in optionality of the types. Developers building a new app will automatically generate code with the latest changes and no action is required.
+
+### **Where do I make these changes?**
+
+1. Update Amplify CLI to the latest version
+
+```
+amplify upgrade
+```
+
+2. The version should be at least x.y.z
+
+```
+amplify --v // at least x.y.z
+```
+
+3. If building an existing app, toggle the feature flag to true in TODO: PATH
+
+4. Run `amplify codegen models` to generate the latest models.
+
+5. Open the App and make sure the app compiles. Depending on your schema, you may be in the following scenarios.
```

Comment:
re: where do developers make these changes? I think for iOS at least, their models are linked in the Xcode project, so it would be under the AmplifyModel folder. I can update this note in the iOS block below that they can view the latest generated models there, and the actually code changes are dependent on their app code

Reply:
The generated models will be in `/amplify/generated/models` folder from the app root. I believe they are linked to AmplifyModel folder, so either paths work.